pkg/virt-controller/watch/migration.go

```diff
@@ -704,8 +710,27 @@ func (c *MigrationController) handleTargetPodCreation(key string, migration *vir
 	// migration was accepted into the system, now see if we
 	// should create the target pod
 	if vmi.IsRunning() {
-		if err := c.createPDB(key, vmi, migration); err != nil {
-			return err
+		if migrations.MigrationNeedsProtection(vmi) {
+			pdbs, err := pdbs.PDBsForVMI(vmi, c.pdbInformer)
+			if err != nil {
+				return err
+			}
+			if len(pdbs) < 1 {
```

Comment:
right here, why not simply filter out the old pdbs that you know you don't want to ever look at in migration.go

something like

```
// removes pdbs from old implementation from list.
pdbs = c.filterOutOldPDBs(pdbs)
```

then you don't have to worry about the 50:50 chance you get the wrong pdb somehow during update. it also avoids an unnecessary error.

Reply:
good point, this is faster as well, PR updated with your suggestion, thank you!